_data-prepper/pipelines/configuration/processors/mutate-event.md

```diff
@@ -0,0 +1,308 @@
+---
+layout: default
+title: Mutate event
+parent: Processors
+grand_parent: Pipelines
+nav_order: 45
+---
+
+# Mutate event processors
+
+You can use mutate event processors to add entries to an event, delete entries from an event, rename keys in an event, copy values within an event, and convert value types in an event. <!---Need a better overall description to explain what they do in general.--->
+
+The following processors allow you to mutate an event:
+
+<!--- Need to find out why links below aren't working correctly.>
+
+* [AddEntry] (#AddEntry)
+* [CopyValue] (#CopyValue)
+* [DeleteEntry] (#DeleteEntry)
+* [RenameKey] (#RenameKey)
+* [ConvertEntry] (#ConvertEntry)
+
+--->
+
+## AddEntry
+
+The `AddEntry` processor adds entries to an event.
+
+### Configuration
+
+You can configure the `AddEntries` processor with the following options:
+
+<!--- Table format.--->
+
+Option | Required | Description
+:--- | :--- | :---
+|`entries` | Yes | A list of entries to add to an event. |
+|`key` | Yes | The key of the new entry to be added. |
+|`value` | Yes | The value of the new entry to be added. You can use the following data types: strings, booleans, numbers, null, nested objects, and arrays. |
+|`overwrite_if_key_exists` | No | When set to `true`, if `key` already exists in the event, then the existing value is overwritten. The default value is `false`. |
+
+### Basic usage
+
+To get started, create the following `pipeline.yaml` file:
+
+```yaml
+pipeline:
+  source:
+    file:
+      path: "/full/path/to/logs_json.log"
+      record_type: "event"
+      format: "json"
+  processor:
+    - add_entries:
+        entries:
+        - key: "newMessage"
+          value: 3
+          overwrite_if_key_exists: true
+  sink:
+    - stdout:
+```
+
+Create a log file named `logs_json.log` and replace the `path` in the file source of your `pipeline.yaml` with this filepath.
```

Comment:
@oeyh Where should you create/store this log file?

Reply:
This log file can basically be put anywhere and then you should specify the path to the log file in the above `path:` field  (i.e. replace "/full/path/to/logs_json.log"). This way Data Prepper will know where to look for that log file.